hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/fees/calculation/SimulatedFcfsUsagePrices.java

```diff
@@ -0,0 +1,199 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import static com.hederahashgraph.api.proto.java.SubType.DEFAULT;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.hederahashgraph.api.proto.java.CurrentAndNextFeeSchedule;
+import com.hederahashgraph.api.proto.java.FeeComponents;
+import com.hederahashgraph.api.proto.java.FeeData;
+import com.hederahashgraph.api.proto.java.FeeSchedule;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.SubType;
+import com.hederahashgraph.api.proto.java.Timestamp;
+import com.hederahashgraph.api.proto.java.TimestampSeconds;
+import com.hederahashgraph.api.proto.java.TransactionFeeSchedule;
+import java.time.Instant;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import javax.inject.Singleton;
+import lombok.RequiredArgsConstructor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.hedera.mirror.common.domain.entity.EntityId;
+import com.hedera.mirror.common.domain.entity.EntityType;
+import com.hedera.mirror.common.domain.file.FileData;
+import com.hedera.mirror.web3.repository.FileDataRepository;
+import com.hedera.services.transaction.fees.calculation.UsagePricesProvider;
+import com.hedera.services.transaction.pricing.RequiredPriceTypes;
+
+@Singleton
+@RequiredArgsConstructor
+public class SimulatedFcfsUsagePrices implements UsagePricesProvider {
+    private static final Logger log = LogManager.getLogger(
+            SimulatedFcfsUsagePrices.class);
+
+    private static final EntityId FEE_SCHEDULE_ENTITY_ID = new EntityId(0L, 0L, 111L, EntityType.FILE);
+
+    private static final long DEFAULT_FEE = 100_000L;
+    private static final FeeComponents DEFAULT_PROVIDER_RESOURCE_PRICES = FeeComponents.newBuilder()
+            .setMin(DEFAULT_FEE)
+            .setMax(DEFAULT_FEE)
+            .setConstant(0).setBpt(0).setVpt(0).setRbh(0).setSbh(0).setGas(0).setTv(0).setBpr(0).setSbpr(0)
+            .build();
+
+    public static final Map<SubType, FeeData> DEFAULT_RESOURCE_PRICES = Map.of(DEFAULT, FeeData.newBuilder()
+            .setNetworkdata(DEFAULT_PROVIDER_RESOURCE_PRICES)
+            .setNodedata(DEFAULT_PROVIDER_RESOURCE_PRICES)
+            .setServicedata(DEFAULT_PROVIDER_RESOURCE_PRICES)
+            .build());
+
+    CurrentAndNextFeeSchedule feeSchedules;
+
+    private Timestamp currFunctionUsagePricesExpiry;
+    private Timestamp nextFunctionUsagePricesExpiry;
+
+    private EnumMap<HederaFunctionality, Map<SubType, FeeData>> currFunctionUsagePrices;
+    private EnumMap<HederaFunctionality, Map<SubType, FeeData>> nextFunctionUsagePrices;
+
+    private FileDataRepository fileDataRepository;
```

Comment:
Won't be injected unless part of the constructor.

```suggestion
    private final FileDataRepository fileDataRepository;
```